lib/test/clojure_lsp/refactor/transform_test.clj

```diff
@@ -303,6 +303,82 @@
     (is (= "^:private a" (cycle-privacy "(defn |a [])")))
     (is (= "a" (cycle-privacy "(defn ^:private |a [])")))))
 
+(defn cycle-fn-literal [code]
+  (as-string (transform/cycle-fn-literal (h/zloc-from-code code))))
+
+(deftest cycle-fn-literal-test
+  (testing "literal to fn"
+    (are [expected fn-literal] (= expected (cycle-fn-literal fn-literal))
+      ;; basic params
+      "(fn [])"                                                      "|#()"
+      "(fn [] (+ 1 2))"                                              "|#(+ 1 2)"
+      "(fn [element] (+ 1 element))"                                 "|#(+ 1 %)"
+      "(fn [element1] (+ 1 element1))"                               "|#(+ 1 %1)"
+      "(fn [element1 element2] (+ 1 element1 element2))"             "|#(+ 1 %1 %2)"
+      ;; mixed numbering styles
+      "(fn [element] (+ 1 element element))"                         "|#(+ 1 % %1)"
+      "(fn [element element2] (+ 1 element element2))"               "|#(+ 1 % %2)"
+      ;; vararg
+      "(fn [& args] (+ 1 args))"                                     "|#(+ 1 %&)"
+      "(fn [element & args] (+ 1 element args))"                     "|#(+ 1 % %&)"
+      "(fn [element1 element2 & args] (+ 1 element1 element2 args))" "|#(+ 1 %1 %2 %&)"
```

Comment:
should we have a test for fn with restructuring? we will probably not be able to cycle for those cases right? maybe just don't return the code action for those cases? WDYT?

Reply:
> not be able to cycle for those cases

Correct. I think there's already a test for that. See `can-cycle-fn-literal-test`.